developer/frameworks/articles/data-engine/01-data-definition-api/01-data-definition-api-operations.markdown

```diff
@@ -94,10 +105,85 @@ To delete a `DEDataDefinition`, pass a `DEDataDefinitionDeleteRequest` to
 Required parameter: `long dataDefinitionId`
 Optional parameters: none
 
-<!-- API NOT READY FOR DOCS YET -->
-<!--
-## Defining Permissions on Data Definitions
+## Granting and Revoking Permissions on Data Definitions
+
+The permissions operations for `DEDataDefinition`s work similarly to the other
+operations. Pass a `DEDataDefinition*PermissionsRequest` to
+`DEDataDefinitionService.execute`. There are permissions for granting and
+revoking permissions on Data Definitions.
+
+### Granting and Revoking Permission to Add Data Definitions and Manage Permissions
+
+`DEDataDefinitionSavePermissionsRequest` is for granting one or more Roles
+permission to add a new Data Definition to the system, and/or the ability to
+manage User permissions on Data Definitions.
+
+Required parameters: 
+- `long companyId`
+- `long scopeGroupId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Optional builder methods:
+- `allowAddDefinition()` to grant Add permissions
+- `allowDefinePermissions()` to grant permissions to grant permissions
```

Comment:
this specification could be: 
"allowDefinePermissions(): the user will be allowed to grant other users with permission."
But if you choose not to edit this part, it's okay to me 👍

Reply:
@marcelabc can the Role User with this permission grant permission to manage Model Permissions or to manage Add and DefinePermission? Also, do they grant this to a Role?

Reply:
The user granted with the permission "DefinePermission" can grant another user with Model Permissions and/or Add/DefinePermission. Besides that, a user can only use the Delete Permission/Model Permission if has the "Define Permission". 

It`s a lot of detail, right?  :)

Reply:
It is a lot of detail! Let's make sure I have it right:
Let's say we give RoleA the allowDefinePermissions permission but not the allowAddDefinitions permission.

That means a User in RoleA can now
1. Give another Role permission to view, update, or delete a specific data definition
2. Give another Role permission to grant other Roles permissions.
3. Delete a Data Definition IF they also have the delete Model Permission for that defintiionId.

But that User cannot
1. Add Data Definitions.
2. Grant other Roles Add Definitions permission.

Is that right?

Reply:
You are almost there! ;)
 
User in RoleA can now:


1. Give another Role permission to view, update, or delete a specific data definition
2. Give another Role permission to DefinePermission
3. Give another Role permission to AddDataDefinition
4. Revoke permission (definePermission/AddDataDefinition)
5. Revoke model permission (delete/update/view)


The user in RoleA can only perform action as add/delete/update/view if his role receives the permission.

Reply:
Okay, I think I have it. I just find it interesting and a little confusing that the User in RoleA cannot Add Data Definitions but can give another Role permission to Add Data Definitions.

Reply:
Oh I see, so he could grant himself the Add permission too. That sounds fair :)